### ISSUE TITLE
Allow writting of the save file during scan, not after

### ISSUE THREAD

**Issue by beaster99** [Lazza/RecuperaBit#65]
The current implementation seems to defer writing the save file to disk after a scan process has completed. There is no method to view this file if you need to cancel or suspend the scan for some reason. This might present other methods for dealing with resuming scans at a specific last know point during the process.

**Comment by beaster99** [comment]
The tool does not seem to export log information outside of what is exported during the save file write process. This means that if there is some issue during a scan, it's not feasible to seek support from the team/community.

**Comment by Lazza** [comment]
I agree, this is quite related to #46, please also see my comment there. 👍 

> it's not feasible to seek support from the team/community

I would like to share your optimism, but currently the "team" is myself (when I have some free time, and currently not much) while I don't think there is quite a community of RecuperaBit users. 😬
